File: WebsiteTinTuc.Admin/vue/src/store/entities/company.ts
import IObjectFile from "../interfaces/IObjectFile";
import Post from "./post";
import BranchJobCompany from "./branch-job-company";
import Entity from "./entity";
import { Guid } from "guid-typescript";

export default class Company extends Entity<Guid> {
    name: string;
    creationTime: Date;
    description: string;
    phone: string;
    email: string;
    locationDescription: string;
    location: string;
    fullNameCompany: string;
    website: string;
    minScale?: number;
    maxScale?: number;
    treatment: string;
    isHot: boolean;
    nationalityId: string;
    nationalityCompanyName: string;
    userId: number;
    creatorName: string;
    thumbnail?: IObjectFile;
    nationalityImage?: IObjectFile;
    images?: Array<IObjectFile>;
    hashtags?: Array<HashtagModel>;
    posts?: Array<Post>;
    branchJobCompanies?: Array<BranchJobCompanyModel>;
}

export class HashtagModel extends Entity<Guid> {
    companyId: string;
    hashtagId: string;
    companyName: string;
    hashtagName: string;
}

export class BranchJobCompanyModel extends Entity<Guid> {
    companyId: string;
    branchJobId: string;
    companyName: string;
    branchJobName: string;
}

File: WebsiteTinTuc.Admin/aspnet-core/src/WebsiteTinTuc.Admin.Core/Entities/Company.cs
using Abp.Domain.Entities.Auditing;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using WebsiteTinTuc.Admin.Authorization.Users;

namespace WebsiteTinTuc.Admin.Entities
{
    public class Company : FullAuditedEntity<Guid>
    {
        [Required]
        public string Name { get; set; }
        public string Description { get; set; }
        public string Phone { get; set; }
        public string Email { get; set; }
        public string LocationDescription { get; set; }
        public string Location { get; set; }
        public string FullNameCompany { get; set; }
        public string Website { get; set; }
        public int? MinScale { get; set; }
        public string Treatment { get; set; }
        public int? MaxScale { get; set; }
        public bool IsHot { get; set; }
        public string CompanyUrl { get; set; }
        [ForeignKey(nameof(Nationality))]
        public Guid NationalityId { get; set; }
        public virtual Nationality Nationality { get; set; }
        [ForeignKey(nameof(User))]
        public long UserId { get; set; }
        public virtual User User { get; set; }
        public virtual ICollection<Post> Posts { get; set; }
        public virtual ICollection<Asset> Assets { get; set; }
        public virtual ICollection<CompanyPostHashtag> CompanyPostHashtags { get; set; }
        public virtual ICollection<BranchJobCompany> BranchJobCompanies { get; set; }
    }
}
